core/src/com/unciv/ui/screens/civilopediascreen/CivilopediaSearchPopup.kt

```diff
@@ -0,0 +1,178 @@
+package com.unciv.ui.screens.civilopediascreen
+
+import com.badlogic.gdx.scenes.scene2d.Actor
+import com.badlogic.gdx.scenes.scene2d.ui.Cell
+import com.badlogic.gdx.scenes.scene2d.ui.SelectBox
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton
+import com.badlogic.gdx.utils.Align
+import com.unciv.models.ruleset.Ruleset
+import com.unciv.models.ruleset.RulesetCache
+import com.unciv.models.stats.INamed
+import com.unciv.models.translations.tr
+import com.unciv.ui.components.ExpanderTab
+import com.unciv.ui.components.UncivTextField
+import com.unciv.ui.components.extensions.disable
+import com.unciv.ui.components.extensions.enable
+import com.unciv.ui.components.extensions.toLabel
+import com.unciv.ui.components.input.KeyCharAndCode
+import com.unciv.ui.components.input.onClick
+import com.unciv.ui.popups.Popup
+import com.unciv.ui.popups.ToastPopup
+import com.unciv.ui.screens.basescreen.BaseScreen
+import com.unciv.ui.screens.basescreen.TutorialController
+import com.unciv.utils.Concurrency
+import com.unciv.utils.launchOnGLThread
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.isActive
+import com.badlogic.gdx.utils.Array as GdxArray
+
+class CivilopediaSearchPopup(
+    private val pediaScreen: CivilopediaScreen,
+    private val tutorialController: TutorialController,
+    private val linkAction: (String) -> Unit
+) : Popup(pediaScreen) {
+    private var ruleset = pediaScreen.ruleset
+    private val searchText = UncivTextField.create("Search text")
+    private val modSelect = ModSelectBox()
+    private lateinit var resultExpander: ExpanderTab
+    private val resultCell: Cell<Actor?>
+    private val searchButton: TextButton
+
+    private var searchJob: Job? = null
+    private var checkLine: (String) -> Boolean = { _ -> false }
+
+    init {
+        searchText.maxLength = 100
+
+        add("{Search text}:".toLabel())
+        add(searchText).growX().row()
+        add("Mod filter:".toLabel())
```

Comment:
translation values. Also, why the difference? Either both should get the : as part of the original text, or neither